app/javascript/components/ApplicationItem.tsx

```diff
@@ -0,0 +1,138 @@
+          </p>
+        )}
+      </header>
+      <section className={"application-item__content"}>
+        <div className="space-y-2">
+          <p className={"application-item__text"}>{props.listingAddress}</p>
+          {props.confirmationNumber && props.submitted && (
+            <div className={"application-item__confirm-text"}>
+              {t("myApplications.yourLotteryNumberIs", {
+                lotteryNumber: props.confirmationNumber,
+              })}
+            </div>
+          )}
+        </div>
+        <div className={"application-item__action"}>
+          <p className={"application-item__status"}>
+            {t("t.status") + ": "}
+            {props.submitted && !props.lotteryComplete && (
+              <span className={"submitted"}>{t("t.submitted")}</span>
+            )}
+            {!props.submitted && !props.pastDue && t("t.inProgress")}
+            {props.submitted && props.lotteryComplete && (
+              <span className={"submitted"}>{t("myApplications.resultsPosted")}</span>
+            )}
+            {!props.submitted && props.pastDue && (
+              <span className={"never-submitted"}>{t("t.neverSubmitted")}</span>
+            )}
+          </p>
+          {props.submitted && !props.lotteryComplete && (
+            <Button
+              onClick={() =>
+                (window.location.href = getLocalizedPath(
+                  props.applicationURL,
+                  getCurrentLanguage()
+                ))
+              }
+            >
+              {t("label.viewApplication")}
+            </Button>
+          )}
+          {props.submitted && props.lotteryComplete && props.lotteryError && (
+            <Button
+              className={AppearanceStyleType.primary}
+              onClick={() =>
+                (window.location.href = getLocalizedPath(
+                  props.lotteryResultsURL,
+                  getCurrentLanguage()
+                ))
+              }
+            >
+              {t("listings.downloadLotteryResults")}
+            </Button>
+          )}
+          {props.submitted && props.lotteryComplete && !props.lotteryError && (
+            <Button
+              className={AppearanceStyleType.primary}
+              onClick={() =>
+                (window.location.href = getLocalizedPath(
+                  props.applicationURL,
+                  getCurrentLanguage()
+                ))
+              }
+            >
+              {t("listings.viewLotteryResults")}
+            </Button>
+          )}
+          {!props.submitted && !props.pastDue && (
+            <Button
+              className={AppearanceStyleType.primary}
+              onClick={() =>
+                (window.location.href = getLocalizedPath(
+                  props.applicationURL,
+                  getCurrentLanguage()
+                ))
+              }
+            >
+              {t("label.continueApplication")}
+            </Button>
+          )}
```

Comment:
suggestion: You can DRY things up like this, it might solve the codeclimate issue:
```javascript
const MyButton = (url, text) => 
  <Button
      className={AppearanceStyleType.primary}
      onClick={() =>
        (window.location.href = getLocalizedPath(
          url,
          getCurrentLanguage()
        ))
      }
    >
    {text}
  </Button>
```